client/state/jetpack-connection-health/actions.js

```diff
@@ -37,3 +31,56 @@ export const setJetpackConnectionUnhealthy = ( siteId, errorCode ) => ( {
 	siteId,
 	errorCode,
 } );
+
+/**
+ * Requests the Jetpack connection status from the server
+ * @param {number} siteId The site id to which the status belongs
+ * @returns {Function} Action thunk
+ */
+export const requestJetpackConnectionHealthStatus = ( siteId ) => ( dispatch, getState ) => {
+	const currentState = getState();
+	const lastRequestTime = currentState.jetpackConnectionHealth[ siteId ]?.lastRequestTime;
+	if ( lastRequestTime && Date.now() - lastRequestTime < 1000 * 60 * 5 ) {
+		return currentState?.jetpackConnectionHealth[ siteId ]?.connectionHealth?.is_healthy;
+	}
+
+	const reduxIsUnhealthy = isJetpackConnectionUnhealthy( currentState, siteId );
+
+	dispatch( {
+		type: JETPACK_CONNECTION_HEALTH_REQUEST,
+		siteId,
+		lastRequestTime: Date.now(),
+	} );
+
+	return wp.req
+		.get( {
+			path: `/sites/${ siteId }/jetpack-connection-health`,
+			apiNamespace: 'wpcom/v2',
+		} )
+		.then( ( response ) => {
+			const { isHealthy, error } = response;
+			if ( isHealthy && reduxIsUnhealthy ) {
+				dispatch( setJetpackConnectionHealthy( siteId ) );
+			}
+			if ( ! isHealthy && ! reduxIsUnhealthy ) {
+				dispatch( setJetpackConnectionUnhealthy( siteId, error ?? '' ) );
+			}
+		} )
+		.catch( ( error ) => {
+			dispatch( {
+				type: JETPACK_CONNECTION_HEALTH_REQUEST_FAILURE,
```

Comment:
In case where jetpack connection endpoint returns a healthy response, and the health status request fails, we will keep showing the error message to the user instead of not showing him anything. This is why we check that inside `JetpackConnectionHealthBanner` component.